### PR TITLE
Port usage of client-go to v3. 

### PR DESCRIPTION
**Pull request by davidvossel**
Client-go 1.5 is being depreciated. The new v3 includes some improved abstractions for using shared informers as well.

### CODE REVIEW

File: cmd/virt-handler/virt-handler.go [hunk 1]
```diff
@@ -88,7 +89,8 @@ func main() {
 	}
 	broadcaster := record.NewBroadcaster()
 	broadcaster.StartRecordingToSink(&kubecorev1.EventSinkImpl{Interface: coreClient.Events(api.NamespaceDefault)})
-	recorder := broadcaster.NewRecorder(kubev1.EventSource{Component: "virt-handler", Host: *host})
+	// TODO what is scheme used for in Recorder?
+	recorder := broadcaster.NewRecorder(kubeapi.Scheme, kubev1.EventSource{Component: "virt-handler", Host: *host})
```
Comment (davidvossel) [inline cmd/virt-handler/virt-handler.go]: Out of all these changes, this is the only one I didn't quite understand.  The recorder takes in the scheme as an argument now. I'm not entirely sure what I did here makes sense.

**Comment by davidvossel** [comment]
Hey, so about vendors.json

Client-go has logic split out into apimachinery now. These two projects are tightly coupled, so a release of client-go expects an exact version of apimachinery.  The versions I used in the vendor.json file are the ones associated with the v3 beta.0 release (https://github.com/kubernetes/client-go/commit/3627aeb7d4f6ade38f995d2c923e459146493c7e)

If we were using godep instead of govendor, I believe godep would have flattened out the apimachinery dependency required by client-go into the kubevirt vendors directory. I couldn't figure out how to get govendor to do the same thing so I explicitly defined the apimachinery version coupled to the client-go version.

Client-go also requires glog to be explicitly pulled into our vendors folder now.

Here's a reference with what's going on there. https://github.com/kubernetes/client-go/blob/master/INSTALL.md

As for the other dependencies that were pulled into vendors.json, I need to investigate why that happened. That wasn't intended.

**Comment by rmohr** [comment]
Hm, I think on the long run, a switch to [Glide](https://github.com/Masterminds/glide) makes most sense. It will do the flattening and can understand godep dependencies.

**Comment by davidvossel** [comment]
Updated the patch.
- fixed vendors.json
- confirmed functional tests pass

**Pull request merged**